### PR TITLE
Configure caching for workflow "Rust"

### PR DESCRIPTION
**Pull request by NobodyXu**
Signed-off-by: Jiahao XU <Jiahao_XU@outlook.com>

### CODE REVIEW

**Comment by NobodyXu** [comment]
Adding caching significantly speedup the CI.
It mostly avoids rebuilding graphicsmagick, which takes roughly 2m and might speedup cargo rebuild.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -103,6 +116,19 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Generate Cargo.lock
+        run: cargo +nightly update
```
Comment (jmjoy) [inline .github/workflows/rust.yml:120]: Why use nightly here?

Comment (NobodyXu) [reply .github/workflows/rust.yml:120]: I use `CARGO_UNSTABLE_SPARSE_REGISTRY: "true"` to speedup fetching of crates.io
It cuts the fetching time from 30s to 1s, eliminating the need of git cloning.

**Review by jmjoy — APPROVED** [review]

**Comment by NobodyXu** [comment]
> Do you have anything else to change? I'm ready to release a new version.

Yes, I also have one PR to go, it's pretty trivial.

**Comment by jmjoy** [comment]
> > Do you have anything else to change? I'm ready to release a new version.
> 
> Yes, I also have one PR to go, it's pretty trivial.

:ok_hand:

**Comment by NobodyXu** [comment]
@jmjoy Never mind the PR, you can release it right now.

**Comment by jmjoy** [comment]
> @jmjoy Never mind the PR, you can release it right now.

No hurry. I can wait for you.

**Comment by NobodyXu** [comment]
> No hurry. I can wait for you.

It's ok, I don't think I have anything to add right now.

**Comment by NobodyXu** [comment]
@jmjoy I do have something in mind, but I don't know how to implement it yet.
It would post it this evening and it would not contain any breaking change.

**Comment by jmjoy** [comment]
Don't worry about break change, just play to your heart's content.

**Pull request merged**